modular_skyrat/modules/customization/modules/clothing/storage/belts.dm

```diff
@@ -0,0 +1,115 @@
+	if(!user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY, FALSE, TRUE))
+		return
+	if(contents.len == 2)
+		var/obj/item/drawn_item = contents[2]
+		add_fingerprint(user)
+		playsound(src, 'sound/items/unsheath.ogg', 50, TRUE, -5)
+		if(!user.put_in_hands(drawn_item))
+			to_chat(user, "<span class='notice'>You fumble for [drawn_item] and it falls on the floor.</span>")
+			update_appearance()
+			return
+		update_appearance()
+		user.visible_message("<span class='notice'>[user] takes [drawn_item] out of [src].</span>", "<span class='notice'>You take [drawn_item] out of [src].</span>")
+	else
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+	. = ..()
+
+/obj/item/storage/belt/crusader/update_appearance(updates)
+	if(contents.len == 2)	//Checks for a sword/rod in the sheath slot, changes the sprite accordingly
+		icon_state = "crusader_belt_sheathed"
+		worn_icon_state = "crusader_belt_sheathed"
+	else
+		icon_state = "crusader_belt"
+		worn_icon_state = "crusader_belt"
+	. = ..()
+
+/obj/item/storage/belt/crusader/examine(mob/user)
+	. = ..()
+	.+= "<span class='notice'>Ctrl-click it to easily open its inventory.</span>"
+	if(contents.len == 2)	//If there's no sword/rod in the sheath slot it doesnt display the alt-click instruction
+		. += "<span class='notice'>Alt-click it to quickly draw the blade.</span>"
+		return
+
+/obj/item/storage/belt/crusader/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+
+	STR.max_items = 2
+	STR.rustle_sound = TRUE
+	STR.max_w_class = WEIGHT_CLASS_GIGANTIC	//Max size makes sure that the storage_pouch item still properly fits in it - the whitelist keeps anything fucky out
+	STR.allow_big_nesting = TRUE //Same as above
+	STR.set_holdable(list(
+		/obj/item/storage/belt/storage_pouch,
+		/obj/item/melee/sabre,
+		/obj/item/melee/cleric_mace,
+		/obj/item/nullrod	//holds any subset of nullrod in the sheath-storage, so this doesnt become a massive list
+		))
+
+/obj/item/storage/belt/crusader/PopulateContents()
+	. = ..()
+	new /obj/item/storage/belt/storage_pouch(src)
+	return
+
+/obj/item/storage/belt/storage_pouch	//seperate mini-storage inside the belt, leaving room for only one sword. Inspired by a (very poorly implemented) belt on Desert Rose
+	icon = 'modular_skyrat/modules/customization/icons/obj/clothing/storage.dmi'
+	worn_icon = 'modular_skyrat/modules/customization/icons/mob/clothing/storage.dmi'
+	name = "storage pouch"
+	desc = "<span class='notice'>Click on this to open your belt's inventory!</span>"
+	icon_state = "storage_pouch_icon"
+	worn_icon_state = "no name"	//Intentionally sets the worn icon to an error
+	w_class = WEIGHT_CLASS_GIGANTIC	//Keeps it from being dragged into other bags
```

Comment:
I'm not a fan of this prevention system. Make it so that it's physically stuck, so that no matter what you can't move it?

Reply:
That's actually just outdated from the original solution before Funce's magic, I can just plain remove that. Thanks for pointing it out